src/main/java/net/coderbot/iris/pipeline/DeferredWorldRenderingPipeline.java

```diff
@@ -790,9 +787,12 @@ private void prepareRenderTargets() {
 
 			// TODO: Support shadow clear color directives & disable buffer clearing
 			// Ensure that the color and depth values are cleared appropriately
-			RenderSystem.clearColor(1.0f, 1.0f, 1.0f, 1.0f);
-			RenderSystem.clearDepth(1.0f);
-			RenderSystem.clear(GL11C.GL_DEPTH_BUFFER_BIT | GL11C.GL_COLOR_BUFFER_BIT, false);
+			if (shadowDirectives.getColorSamplingSettings().get(0).getClear()) {
```

Comment:
What does OptiFine do with shadowtex1Clear=true ?

This doesn't appear to handle the separate clear color settings and clear toggle for the color buffers either. You probably need to rework this section of code.